packages/dev/core/src/FlowGraph/flowGraphCoordinator.ts

```diff
@@ -91,12 +92,18 @@ export class FlowGraphCoordinator {
         this._onBeforeRenderObserver = this.config.scene.onBeforeRenderObservable.add(() => {
             // Reset the event execution counter at the beginning of each frame.
             this._eventExecutionCounter.clear();
-            if (this._executeOnNextFrame.length) {
+            // duplicate the _executeOnNextFrame array to avoid modifying it while iterating over it
+            const executeOnNextFrame = this._executeOnNextFrame.slice(0);
+            if (executeOnNextFrame.length) {
                 // Execute the events that were triggered on the next frame.
-                this._executeOnNextFrame.forEach((event) => {
+                executeOnNextFrame.forEach((event) => {
                     this.notifyCustomEvent(event.id, event.data, false);
+                    // remove the event from the array

```

Comment:
For my understanding:

You updated the code because `this.notifyCustomEvent()` can potentially add or remove elements from `this._executeOnNextFrame`?